### PR TITLE
Stekcub

### PR DESCRIPTION
**Pull request by jmframe**
Added a blog post about buckets, of all things. Also added a section in the main page about contributions. Feel free to modify. And added a math plugin.

### CODE REVIEW

File: docs/blog/posts/buckets_as_conceptual_hydr.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+## A walk through the watershed
+
+Going back again to undergraduate, I was building my first watershed model (Frame, 2010). I spend weeks staring at a computer screed, writing code, processing data, and thinking through discretization of Sir Edmond's ale bucket. I went out to do some field work for a different project in the Carmel River Watershed, a valley I've been to a dozen times. But after spending so much time thinking about this watershed digitally, well really just a digital bucket representing the watershed, a sense of dred came over me as I looked down the Carmel Valley (Figured below). I was realizing how much my precious computer model was missing, seeing the errors in my assumptions, the variety of vegetation in areas my models represents as homogeneous, the property boundaries with drainage features. Nothing in the analog system seemed to match my digitization. There is overwhelming complexity in every watershed, river reach and hillslope, if our eyes are open.
```
Comment (taddyb) [inline docs/blog/posts/buckets_as_conceptual_hydr.md:34]: I spend weeks staring at a computer screen (typo)

File: docs/blog/posts/buckets_as_conceptual_hydr.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+Nash, J. E., & HRS. (1960). A unit hydrograph study, with particular reference to British catchments. Proceedings of the Institution of Civil Engineers, 17(3), 249-282.
+
+Nearing, G. S., Kratzert, F., Sampson, A. K., Pelissier, C. S., Klotz, D., Frame, J. M., Prieto, C., & Gupta, H. v. (2020). What Role Does Hydrological Science Play in the Age of Machine Learning? Water Resources Research. https://doi.org/10.1029/2020wr028091
+
+Shen, C., Appling, A. P., Gentine, P., Bandai, T., Gupta, H., Tartakovsky, A., ... & Lawson, K. (2023). Differentiable modelling to unify machine learning and physical models for geosciences. Nature Reviews Earth & Environment, 4(8), 552-567.
+
+Wang, Y. H., & Gupta, H. V. (2024). Towards interpretable physical‐conceptual catchment‐scale hydrological modeling using the mass‐conserving‐perceptron. Water Resources Research, 60(10), e2024WR037224.
+
+<!-- more -->
```
Comment (taddyb) [inline docs/blog/posts/buckets_as_conceptual_hydr.md:101]: We should move this up earlier

**Review by taddyb — CHANGES_REQUESTED** [review]
I'll fix the typo and the `<!--more-->` placement. Otherwise, looks good to me!

**Review by taddyb — APPROVED** [review]

File: docs/blog/posts/buckets_as_conceptual_hydr.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+But in hydrology, we consider representing a delayed flow of water as a bucket a type of physical conceptualization. And as a matter of fact, this goes back a long way to Instantaneous Unit Hydrograph (IUH). In my MS level hydrology class, we learned that the IUH from linear reservoirs turns into a beta distribution (Nash 1960). This is a type of theoretical physical understanding of hydrology through buckets.  
+
+The equation for the Instantaneous Unit Hydrograph (IUH) from linear reservoirs is:  
+
+$$
+q(t) = \frac{t^{n-1} e^{-t/k}}{k^n \Gamma(n)}
+$$
+
+Where q(t) is the flow rate at time \( t \), k is the storage coefficient (related to the delay in each reservoir), n is the number of reservoirs (controls the shape of the hydrograph), Gamma(n) is the gamma function.
```
Comment (RY4GIT) [inline docs/blog/posts/buckets_as_conceptual_hydr.md:51]: Suggest using math symbols for all variables e.g., $q$

File: docs/blog/posts/buckets_as_conceptual_hydr.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+## Taking buckets to their limit
+
+The actual physical representation of a mathematical structure of a system of buckets flowing into each other is the correct physical model of one type of system only, and that is, of course, a system of buckets flowing into each other. Applying this model to a watershed, for instance, is similar to using continuous energy waves in the form of harmonic oscillators to model black body radiation. Yes, it will do in some cases, but we know it is not how the real world behaves.
```
Comment (RY4GIT) [inline docs/blog/posts/buckets_as_conceptual_hydr.md:65]: This part is a bit wordy "The actual physical representation of a mathematical structure of a system of buckets flowing into each other is the correct physical model of one type of system only, and that is, of course, a system of buckets flowing into each other."

File: docs/blog/posts/buckets_as_conceptual_hydr.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+We've seen overwhelming evidence in the past few decades that neural networks are more accurate at predicting hydrologic responses than the conceptualization of hydrologic systems as buckets (Nearing et al., 2020). Unless that conceptualization itself can be boosted by a neural network (Shen et al., 2024). What I've always pondered is what would happen if hydrologists acted a bit more like physicists, and rejected demonstratively predictive conceptualizations in favor of those that match observations. There is often a sentiment that the bucket conceptualizations, even as simplifications, are "interpretable", but an interpretation is only usefull if we are honest about them limitations. We still learn the Rayleigh–Jeans law in thermodynamics, but we then understand the limitations of the continuous energy conceptualization and turn to Planck's Law to grasp the quantum reality. We can still learn bucket conceptualizations, but let's not pretend that they are "physical" representations.
```
Comment (RY4GIT) [inline docs/blog/posts/buckets_as_conceptual_hydr.md:73]: The sentence "Unless that conceptualization itself can be ..." is an incomplete sentence. 

Also, I'm a bit confused "demonstratively predictive conceptualization" refers to NN or bucket model? 

If it refers to bucket model, some word containing a bit more essence of "empirical", "conventional", "what people are familiar with," or "deterministic bucket" would be helpful. 

If it refers to a NN, some terms inferring "simply provides accurate predictions" or "simply a large network of bucket models in the end" would be helpful.

File: docs/blog/posts/buckets_as_conceptual_hydr.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+The real beauty lies in the interplay between abstraction and application. While we recognize the limitations of these analogies—no bucket, spring, or reservoir can capture the full complexity of a river basin or a turbulent atmosphere—they still provide an essential scaffold for exploration. From conceptual hydrological models to modern machine learning architectures, these analogies offer a common language to connect diverse fields of study, sparking innovation in ways that would be impossible without simplification.  
+
+## References
+Acuña Espinoza, E., Loritz, R., Álvarez Chaves, M., Bäuerle, N., and Ehret, U.: To bucket or not to bucket? Analyzing the performance and interpretability of hybrid hydrological models with dynamic parameterization, Hydrol. Earth Syst. Sci., 28, 2705–2719, https://doi.org/10.5194/hess-28-2705-2024, 2024.
```
Comment (RY4GIT) [inline docs/blog/posts/buckets_as_conceptual_hydr.md:86]: I think you can still cite this somewhere in the main body of text (currently it is not)

File: docs/blog/posts/buckets_as_conceptual_hydr.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+The real beauty lies in the interplay between abstraction and application. While we recognize the limitations of these analogies—no bucket, spring, or reservoir can capture the full complexity of a river basin or a turbulent atmosphere—they still provide an essential scaffold for exploration. From conceptual hydrological models to modern machine learning architectures, these analogies offer a common language to connect diverse fields of study, sparking innovation in ways that would be impossible without simplification.  
+
+## References
+Acuña Espinoza, E., Loritz, R., Álvarez Chaves, M., Bäuerle, N., and Ehret, U.: To bucket or not to bucket? Analyzing the performance and interpretability of hybrid hydrological models with dynamic parameterization, Hydrol. Earth Syst. Sci., 28, 2705–2719, https://doi.org/10.5194/hess-28-2705-2024, 2024. 
+
+Frame, J. M. (2010). An integrated surface water-groundwater interaction model for the Carmel River.
+
+Frame J. M., L. Hernandez Rodriguez, and M. Bassiouni (2023). "DeepBucketLab - A Playground for Understanding Deep Learning for Hydrologic Process Representations," DOI: 10.5072/zenodo.7349
```
Comment (RY4GIT) [inline docs/blog/posts/buckets_as_conceptual_hydr.md:90]: Could not access with this DOI

File: docs/blog/posts/buckets_as_conceptual_hydr.md (hunk 3 above)
Comment (RY4GIT) [reply docs/blog/posts/buckets_as_conceptual_hydr.md]: I made the change in the latest commit

**Review by RY4GIT — APPROVED** [review]
I enjoyed the read!!

File: docs/blog/posts/buckets_as_conceptual_hydr.md (hunk 3 above)
Comment (jmframe) [reply docs/blog/posts/buckets_as_conceptual_hydr.md]: Thanks. I finally figured out how to make it work. Added a markdown_extension:   - pymdownx.arithmatex:
      generic: true

File: docs/blog/posts/buckets_as_conceptual_hydr.md (hunk 4 above)
Comment (jmframe) [reply docs/blog/posts/buckets_as_conceptual_hydr.md:65]: "A mathematical model of buckets flowing into each other accurately represents one specific system: buckets flowing into each other. "

Comment (RY4GIT) [reply docs/blog/posts/buckets_as_conceptual_hydr.md:65]: Hmm I like the previous structure better. How about just removing a first few phrases and then: "The mathematical structure of a system of buckets flowing into each other is the correct physical model of one type of system only, and that is, of course, a system of buckets flowing into each other."

File: docs/blog/posts/buckets_as_conceptual_hydr.md (hunk 3 above)
Comment (RY4GIT) [reply docs/blog/posts/buckets_as_conceptual_hydr.md]: Thank you for adding math plugin!!

File: docs/blog/posts/buckets_as_conceptual_hydr.md (hunk 5 above)
Comment (jmframe) [reply docs/blog/posts/buckets_as_conceptual_hydr.md:73]: Ahh, this: "demonstratively predictive conceptualization" should be "demonstratively poor...". 

It is any model conceptualization.

Re-phrased a bit, I think it should be more clear.

File: docs/blog/posts/buckets_as_conceptual_hydr.md (hunk 6 above)
Comment (jmframe) [reply docs/blog/posts/buckets_as_conceptual_hydr.md:86]: Oh, good catch. I added it assuming I would cite it, but I didn't... okay, let me see. I added this:  The predictive power of bucket-based conceptualizations can themselves be enhanced through integration with neural networks (Shen et al., 2023). So much, in fact, the neural network is more important than the bucket (Acuña Espinoza et al., 2024).

File: docs/blog/posts/buckets_as_conceptual_hydr.md (hunk 5 above)
Comment (RY4GIT) [reply docs/blog/posts/buckets_as_conceptual_hydr.md:73]: I see, then the sentence makes sense now !! 👍🏻

File: docs/blog/posts/buckets_as_conceptual_hydr.md (hunk 7 above)
Comment (jmframe) [reply docs/blog/posts/buckets_as_conceptual_hydr.md]: hmmm, I don't know what is up with that. It is the correct DOI according to Zenodo...? https://sandbox.zenodo.org/records/7349

Comment (RY4GIT) [reply docs/blog/posts/buckets_as_conceptual_hydr.md]: The website says: This is the Zenodo Sandbox instance, used for testing purposes ⚠️
DOIs created in this instance are not real and will not resolve. You can find the production instance of Zenodo at zenodo.org

Comment (jmframe) [reply docs/blog/posts/buckets_as_conceptual_hydr.md]: oh

Comment (jmframe) [reply docs/blog/posts/buckets_as_conceptual_hydr.md]: I guess I didn't finish the process. https://doi.org/10.5281/zenodo.14538195
